oss-internship-2020/openjpeg/examples/decompress_example.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Perform decompression from *.jp2 to *.pnm format
+
+#include <libgen.h>
+#include <syscall.h>
+
+#include <cassert>
+#include <cstdlib>
+#include <iostream>
+
+#include "convert_helper.h"
+#include "openjp2_sapi.sapi.h"
+#include "sandboxed_api/util/flag.h"
+
+class Parameters : public sapi::v::Struct<opj_dparameters_t> {};
+class Opj_image_t : public sapi::v::Struct<opj_image_t> {};
+
+class Openjp2SapiSandbox : public Openjp2Sandbox {
+ public:
+  Openjp2SapiSandbox(const std::string& in_file) : in_file_(in_file) {}
+
+  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
+      sandbox2::PolicyBuilder*) override {
+    return sandbox2::PolicyBuilder()
+        .AllowStaticStartup()
+        .AllowOpen()
+        .AllowRead()
+        .AllowWrite()
+        .AllowStat()
+        .AllowSystemMalloc()
+        .AllowExit()
+        .AllowSyscalls({
+            __NR_futex,
+            __NR_close,
+            __NR_lseek,
+        })
+        .AddFile(in_file_)
+        .BuildOrDie();
+  }
+
+ private:
+  std::string in_file_;
+};
+
+int main(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  if (argc != 3) {
+    std::cerr << "usage: " << basename(argv[0]) << " absolute/path/to/INPUT.jp2"
+              << " absolute/path/to/OUTPUT.pnm\n";
+    return EXIT_FAILURE;
+  }
+
+  std::string in_file(argv[1]);
+
+  // initialize sandbox
+  Openjp2SapiSandbox sandbox(in_file);
+  absl::Status status = sandbox.Init();
+  assert(status.ok());
+
+  Openjp2Api api(&sandbox);
+  sapi::v::ConstCStr in_file_v(in_file.c_str());
+
+  // initialize library's main data-holders
+  sapi::StatusOr<opj_stream_t*> stream_status =
```

Comment:
the current recommendation is to name StausOr's in similar way as you would name var of the underlaying type.
Also don't create another var for the StatusOr's value, but just use `operator*`, `operator->`, or `value()`.